### PR TITLE
OAuth: Remove accessTokenExpirationCheck feature toggle

### PR DESCRIPTION
**Pull request by mgyongyosi**
**What is this feature?**
This PR removes the `accessTokenExpirationCheck` feature toggle completely. The access token expiration check + refresh token fetching can be enabled in the OAuth provider settings using the previously introduced (v10.2) `use_refresh_token` config.

**Why do we need this feature?**

**Who is this feature for?**

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md [hunk 1]
```diff
@@ -99,17 +97,15 @@ The table below describes all GitLab OAuth configuration options. Like any other
...
 
 By default, GitLab provides a refresh token.
 
-Refresh token fetching and access token expiration check is enabled by default for the GitLab provider since Grafana v10.1.0 if the `accessTokenExpirationCheck` feature toggle is enabled. If you would like to disable access token expiration check then set the `use_refresh_token` configuration value to `false`.
+Refresh token fetching and access token expiration check is enabled by default for the GitLab provider since Grafana v10.1.0. If you would like to disable access token expiration check then set the `use_refresh_token` configuration value to `false`.
 
-> **Note:** The `accessTokenExpirationCheck` feature toggle will be removed in Grafana v10.3.0 and the `use_refresh_token` configuration value will be used instead for configuring refresh token fetching and access token expiration check.
+> **Note:** The `accessTokenExpirationCheck` feature toggle has been removed in Grafana v10.3.0 and the `use_refresh_token` configuration value will be used instead for configuring refresh token fetching and access token expiration check.
```
Comment (lwandz13) [inline docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md:108]:
```suggestion
{{% admonition type="note" %}}
The `accessTokenExpirationCheck` feature toggle has been removed in Grafana v10.3.0 and the `use_refresh_token` configuration value will be used instead for configuring refresh token fetching and access token expiration check.
This feature is behind the `accessTokenExpirationCheck` feature toggle.
{{% /admonition %}}
```

Comment (lwandz13) [inline docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md:108]:
```suggestion
{{% admonition type="note" %}}
The `accessTokenExpirationCheck` feature toggle has been removed in Grafana v10.3.0 and the `use_refresh_token` configuration value will be used instead for configuring refresh token fetching and access token expiration check.
{{% /admonition %}}
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/google/index.md [hunk 2]
```diff
@@ -79,17 +79,15 @@ interception attacks. PKCE will be required in [OAuth 2.1](https://datatracker.i
...
 
 By default, Grafana includes the `access_type=offline` parameter in the authorization request to request a refresh token.
 
-Refresh token fetching and access token expiration check is enabled by default for the Google provider since Grafana v10.1.0 if the `accessTokenExpirationCheck` feature toggle is enabled. If you would like to disable access token expiration check then set the `use_refresh_token` configuration value to `false`.
+Refresh token fetching and access token expiration check is enabled by default for the Google provider since Grafana v10.1.0. If you would like to disable access token expiration check then set the `use_refresh_token` configuration value to `false`.
 
-> **Note:** The `accessTokenExpirationCheck` feature toggle will be removed in Grafana v10.3.0 and the `use_refresh_token` configuration value will be used instead for configuring refresh token fetching and access token expiration check.
+> **Note:** The `accessTokenExpirationCheck` feature toggle has been removed in Grafana v10.3.0 and the `use_refresh_token` configuration value will be used instead for configuring refresh token fetching and access token expiration check.
```
Comment (lwandz13) [inline docs/sources/setup-grafana/configure-security/configure-authentication/google/index.md:90]:
```suggestion
{{% admonition type="note" %}}
The `accessTokenExpirationCheck` feature toggle has been removed in Grafana v10.3.0 and the `use_refresh_token` configuration value will be used instead for configuring refresh token fetching and access token expiration check.
{{% /admonition %}}
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/keycloak/index.md [hunk 3]
```diff
@@ -164,10 +164,11 @@ allow_assign_grafana_admin = true
...
 > Available in Grafana v9.3 and later versions.
 
-> **Note:** This feature is behind the `accessTokenExpirationCheck` feature toggle.
-
 When a user logs in using an OAuth provider, Grafana verifies that the access token has not expired. When an access token expires, Grafana uses the provided refresh token (if any exists) to obtain a new access token.
 
 Grafana uses a refresh token to obtain a new access token without requiring the user to log in again. If a refresh token doesn't exist, Grafana logs the user out of the system after the access token has expired.
 
-To enable a refresh token for Keycloak, extend the `scopes` in `[auth.generic_oauth]` with `offline_access`.
+To enable a refresh token for Keycloak, do the followings:
```
Comment (lwandz13) [inline docs/sources/setup-grafana/configure-security/configure-authentication/keycloak/index.md:171]:
```suggestion
To enable a refresh token for Keycloak, do the following: 
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/okta/index.md [hunk 4]
```diff
@@ -129,7 +127,7 @@ To enable the `Refresh Token` head over the Okta application settings and:
 
 At the configuration file, extend the `scopes` in `[auth.okta]` section with `offline_access`.
 
-> **Note:** The `accessTokenExpirationCheck` feature toggle will be removed in Grafana v10.3.0 and the `use_refresh_token` configuration value will be used instead for configuring refresh token fetching and access token expiration check.
+> **Note:** The `accessTokenExpirationCheck` feature toggle has been removed in Grafana v10.3.0 and the `use_refresh_token` configuration value will be used instead for configuring refresh token fetching and access token expiration check.
```
Comment (lwandz13) [inline docs/sources/setup-grafana/configure-security/configure-authentication/okta/index.md:130]:
```suggestion
{{% admonition type="note" %}}
The `accessTokenExpirationCheck` feature toggle has been removed in Grafana v10.3.0 and the `use_refresh_token` configuration value will be used instead for configuring refresh token fetching and access token expiration check.
{{% /admonition %}}
```

**Review by lwandz13 — DISMISSED** [review]

**Review by Jguer — APPROVED** [review]

**Review by chri2547 — APPROVED** [review]

**Review by kalleep — APPROVED** [review]
The code part looks good to me.

One suggestion would to cache the check here https://github.com/grafana/grafana/blob/main/pkg/services/authn/authnimpl/sync/oauth_token_sync.go#L66. So we are not doing additonal database call for every request but that can be done in a followup pr

**Pull request merged**